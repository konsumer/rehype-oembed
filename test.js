/* global test expect */
import oembed from './index.js'
import { reporter } from 'vfile-reporter'
import { unified } from 'unified'
import remarkParse from 'remark-parse'
import remarkGfm from 'remark-gfm'
import remarkRehype from 'remark-rehype'
import remarkStringify from 'rehype-stringify'
import { promises as fs } from 'fs'

test('full doc', async () => {
  const contents = await unified()
    .use(remarkParse)
    .use(remarkRehype)
    .use(remarkGfm)
    .use(oembed)
    .use(remarkStringify)
    .process(await fs.readFile('test.md'))

  expect(reporter(contents, { color: false })).toBe('no issues found')
  expect(String(contents)).toMatchSnapshot()
})
