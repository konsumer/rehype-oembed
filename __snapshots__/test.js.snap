// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`full doc 1`] = `
"<h1>Remark oEmbed</h1>
<p>This is a library to add nice embeds for links, when processing HTML or markdown. You can find the library <a href=\\"https://github.com/konsumer/rehype-oembed\\">here on github</a>.</p>
<p>This is a basic demo-page.</p>
<h2>Twitter</h2>
<p>This is a tweet:</p>
<p><blockquote class=\\"twitter-tweet\\"><p lang=\\"en\\" dir=\\"ltr\\">bad children are sent down \\"the slide of no return\\" <a href=\\"https://t.co/GLxna9yeVT\\">pic.twitter.com/GLxna9yeVT</a></p>â€” Lil Bit ðŸŒˆ (@LizerReal) <a href=\\"https://twitter.com/LizerReal/status/1430945065334104065?ref_src=twsrc%5Etfw\\">August 26, 2021</a></blockquote>
<script async src=\\"https://platform.twitter.com/widgets.js\\" charset=\\"utf-8\\"></script>
</p>
<p>This is another tweet:</p>
<p><blockquote class=\\"twitter-tweet\\"><p lang=\\"en\\" dir=\\"ltr\\">6YO: Can I eat a cookie?<br><br>Me: Finish your dinner first<br><br>6YO: My stomach is full except for a circle shaped space</p>â€” Vinod Chhaproo (@Chhapiness) <a href=\\"https://twitter.com/Chhapiness/status/1422326068917284869?ref_src=twsrc%5Etfw\\">August 2, 2021</a></blockquote>
<script async src=\\"https://platform.twitter.com/widgets.js\\" charset=\\"utf-8\\"></script>
</p>
<h2>Youtube</h2>
<p>This is a youtube video:</p>
<p><iframe width=\\"200\\" height=\\"150\\" src=\\"https://www.youtube.com/embed/K-281doxOMc?feature=oembed\\" frameborder=\\"0\\" allow=\\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture\\" allowfullscreen></iframe></p>
<p><a href=\\"https://www.youtube.com/watch?v=K-281doxOMc\\">this should not match</a></p>
<p>And neither should this: <a href=\\"https://www.youtube.com/watch?v=K-281doxOMc\\">https://www.youtube.com/watch?v=K-281doxOMc</a></p>
<h2>Instagram</h2>
<p>Lil Bub is so stinkin' cute:</p>
<p><a href=\\"https://www.instagram.com/p/CSNAXbgABKw\\">https://www.instagram.com/p/CSNAXbgABKw</a></p>
<p>Unfortunately, Instagram requires a facebook-app + instagram client-id &#x26; approval, which is a pain.</p>
<h2>Flickr</h2>
<p><a data-flickr-embed=\\"true\\" href=\\"https://www.flickr.com/photos/48904813@N04/4480686437/\\" title=\\"cute cat wallpaper04 by nicemonday, on Flickr\\"><img src=\\"https://live.staticflickr.com/4013/4480686437_2d0c7018be_b.jpg\\" width=\\"1024\\" height=\\"768\\" alt=\\"cute cat wallpaper04\\"></a><script async src=\\"https://embedr.flickr.com/assets/client-code.js\\" charset=\\"utf-8\\"></script></p>
<p>That is one cute cat!</p>"
`;
